src/main/java/com/odysseusinc/arachne/executionengine/model/descriptor/r/RExecutionRuntime.java

```diff
@@ -17,14 +19,14 @@ public class RExecutionRuntime implements ExecutionRuntime {
     @JsonProperty
     private String version;
     @JsonProperty
-    private List<RDependency> dependencies = new ArrayList<>();
+    private RDependency[] dependencies = new RDependency[0];
 
     public boolean matches(ExecutionRuntime otherRuntime) {
         if (!(otherRuntime instanceof RExecutionRuntime))
             return false;
         RExecutionRuntime rExecutionRuntime = (RExecutionRuntime) otherRuntime;
         return this.version.equals(rExecutionRuntime.version)
-                && this.dependencies.containsAll(rExecutionRuntime.dependencies);
+                && Arrays.asList(this.dependencies).containsAll(Arrays.asList(rExecutionRuntime.dependencies));
```

Comment:
containsAll may have poor performance, but let's leave it, due to not break an original logic